compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -511,11 +511,18 @@ object Trees {
   }
 
   /** selector match { cases } */
-  case class Match[-T >: Untyped] private[ast] (selector: Tree[T], cases: List[CaseDef[T]])
+  case class Match[-T >: Untyped] private[ast] (selector: Tree[T], cases: List[CaseDef[T]])(val kind: MatchKind)
```

Comment:
Do we really want `Match(s, cs)(Regular) == Match(s, cs)(Inline)`?

Reply:
Tree `==` is identity, so this won't be a problem.